mypy/semanal.py

```diff
@@ -1944,7 +1944,7 @@ def self_type(typ: TypeInfo) -> Union[Instance, TupleType]:
 @overload
 def replace_implicit_first_type(sig: Callable, new: Type) -> Callable:
     # We can detect implicit self type by it having no representation.
-    if not sig.arg_types[0].repr:
+    if True: # TODO: repr field was removed
```

Comment:
This is of course not correct. There is currently not test for it. I'll try to add tests and fix this later.